modules/sdp/LocalSdpMunger.js

```diff
@@ -364,13 +365,21 @@ export default class LocalSdpMunger {
 
         for (const source of sources) {
             const nameExists = mediaSection.ssrcs.find(ssrc => ssrc.id === source && ssrc.attribute === 'name');
+            const msid = mediaSection.ssrcs.find(ssrc => ssrc.id === source && ssrc.attribute === 'msid')?.value;
+            let sourceName = null;
 
+            if (msid) {
+                const msidInfo = msid.split('-');
+
+                msidInfo.splice(3);
+                sourceName = msidInfo.join('-');
```

Comment:
why does source name come from the msid?

Reply:
I am trying to extract the trackIndex to be used for source name generation since we don't have a reference to the track object here. I modified it slightly now.